src/components/Controls/ToggleVideoButton/ToggleVideoButton.tsx

```diff
@@ -19,14 +19,22 @@ const useStyles = makeStyles((theme: Theme) =>
 export default function ToggleVideoButton(props: { disabled?: boolean }) {
   const classes = useStyles();
   const [isVideoEnabled, toggleVideoEnabled] = useLocalVideoToggle();
+  const lastClickTimeRef = useRef(0);
+
+  const toggleVideo = useCallback(() => {
+    if (Date.now() - lastClickTimeRef.current > 200) {
```

Comment:
I think we should use `lodash.throttle`. Same way how you did it here https://github.com/twilio/twilio-video-app-react/blob/master/src/components/Controls/useIsUserActive/useIsUserActive.ts

Reply:
It isn't really possible to use `throttle` in this instance. For example, this doesn't work:

```
export default function ToggleVideoButton(props: { disabled?: boolean }) {
  const classes = useStyles();
  const [isVideoEnabled, toggleVideoEnabled] = useLocalVideoToggle();

  const toggleVideo = useCallback(throttle(toggleVideoEnabled, 200), [toggleVideoEnabled]);

  return (
    <Tooltip
      title={isVideoEnabled ? 'Mute Video' : 'Unmute Video'}
      placement="top"
      PopperProps={{ disablePortal: true }}
    >
      <Fab className={classes.fab} onClick={toggleVideo} disabled={props.disabled}>
        {isVideoEnabled ? <Videocam /> : <VideocamOff />}
      </Fab>
    </Tooltip>
  );
}
```

That's because calling `toggleVideoEnabled` will cause `useCallback` to produce a new function (because the `useLocalVideoToggle` creates new `isVideoEnabled` and `toggleVideoEnabled` variables whenever the local video is toggled). So `throttle` ends up being called everytime the button is clicked, which means that the throttle effect is lost because `onClick` calls a new function each time instead of the same function as before.

The example you provided works fine because `useEffect` runs only once (note the empty dependency array). There are some `useThrottle` packages that can be installed from NPM, but I found it easy enough just to do it myself. If we need to add this functionality to other components, then I'll can add my own `useThrottle` hook to the codebase.

Reply:
Ok 😞